lib/python/ray/worker.py

```diff
@@ -526,6 +629,11 @@ def process_task(task): # wrapping these lines in a function should cause the lo
         (function, arg_types, return_types) = pickling.loads(function)
         if function.__module__ is None: function.__module__ = "__main__"
         worker.register_function(remote(arg_types, return_types, worker)(function))
+      if variable_info is not None:
+        name, initializer_str, reinitializer_str = variable_info
+        initializer = pickling.deserialize(initializer_str)
```

Comment:
They actually won't ever be the same function because `initializer` takes no arguments, and `reinitializer` takes one argument.